docs/docs/04-Configurations/01-basic-configuration.md

```diff
@@ -85,6 +85,52 @@ const config = {
 When making this setting, make sure that the file names (including the path part) of different images do not cover each other.  
 Specifically, include the name, width, quality, and extension in the return value. If path is not included, all src's should be specified with `import` or `require` so that they can be distinguished by their hash value even if they have the same filename.
 
+### `sourceImageParser`
+
+- Type: function
+- Argument: Object
+- Return value: string
+
+The argument for this function will be an object of type
+
+```typescript
+{
+  src: string // The source images 'src' attribute
+  defaultParser: (src: string) => ParsedImageInfo // A function which evaluates the image name, path name (without image name appended and starting w/ '/'), and extension
+}
+```
+
+This might be useful if any of your images have URLs that do not follow the standard `https://somehost.com/imagename.extension` pattern.
+
+For example: Maybe your source image's src attribute is more like `https://somedigitalassetmangementhost.com?fileId=1234-xyze&extension=jpg`, so you might add the following to your config:
+
+**NOTE**
+This gets run before filenameGenerator, so the arguments passed into filenameGenerator would be affected by sourceImageParser configuration. (path, name, and extension)
+
+```typescript
+// export-images.config.js
+/**
+ * @type {import('../../src').Config}
```

Comment:
I would like to present the configurations for general users in the document, so the following is a good place to start.

```diff
- * @type {import('../../src').Config}
+ * @type {import('next-export-optimize-images').Config}
```